### ISSUE TITLE
Model not saving?

### ISSUE THREAD

**Issue by youssefavx** [UKPLab/sentence-transformers#708]
Hey guys, currently trying to fine tune but after fine tuning, I don't see the model in the directory I pointed to. Here's my code:

```
from sentence_transformers import SentenceTransformer,  SentencesDataset, LoggingHandler, losses
from sentence_transformers.readers import InputExample
from torch.utils.data import DataLoader
from sentence_transformers import evaluation

train_examples = []
first_10_examples = csvlist[:10]
for sentence_one, sentence_two, score in csvlist[10:]:
  train_examples.append(InputExample(texts=[sentence_one, sentence_two], label=int(score)))

model = SentenceTransformer('stsb-bert-large')
train_dataset = SentencesDataset(train_examples, model)
train_batch_size=1
train_dataloader = DataLoader(train_dataset, shuffle=True, batch_size=train_batch_size)

train_loss = losses.OnlineContrastiveLoss(model=model)
sentences1 = []
sentences2 = []
scores = []
for first_sentence, second_sentence, score in first_10_examples:
  sentences1.append(first_sentence)
  sentences2.append(second_sentence)
  scores.append(score)

evaluator = evaluation.EmbeddingSimilarityEvaluator(sentences1, sentences2, scores)
num_epochs=1
warmup_steps=10
model_save_path='/content/drive/MyDrive'

model.fit(train_objectives=[(train_dataloader, train_loss)],
          #evaluator=evaluator,
          epochs=num_epochs,
          evaluation_steps=1000,
          warmup_steps=warmup_steps,
          output_path=model_save_path)
```

I'm assuming the model is saved in `model_save_path`? 

I was trying to train on Colab, so I tried to save my model in my drive. I also tried saving it in a directory inside my drive, and I also tried to save it in '/content/testing' and still I see no files.

**Comment by nreimers** [comment]
Hi @youssefavx 
model_save_path is only used when you have an evaluator and when a new best score is returned by that evaluator.

In your case, evaluator is None, so no model is saved.

You can save it like this:
```
model.fit(...)
model.save(model_save_path)
```

**Comment by youssefavx** [comment]
@nreimers Ah! Thanks so much for the clarification. I'll try that.

Yeah, I was getting a `RuntimeWarning: invalid value encountered in true_divide` with the evaluator so i commented it out. I'm assuming that's due to the evaluator being cosine and the loss being contrastive or something. I wasn't sure what evaluator to use with contrastive loss.

**Comment by nreimers** [comment]
Hi @youssefavx 
The evaluator is independent of the loss. I.e., you can use any evaluator with any loss.

But you have to pass the right data to the evaluator.

**Comment by youssefavx** [comment]
Oh okay that makes sense. I'll try to just use the one I'm using currently but do something someone had done before like so for zeros `np.random.uniform(low=0.0, high=0.000001)` and like so for ones: ` np.random.uniform(low=0.0, high=0.000001)` and see if that works.
